### PR TITLE
[SPARK-34628][SQL] Remove GlobalLimit operator if its child max rows not larger than limit number

### PR DESCRIPTION
**Pull request by wangyum**
### What changes were proposed in this pull request?

This pr remove `GlobalLimit` operator if its child max rows not larger than limit number. For example:
```
val testRelation = LocalRelation.fromExternalRows(Seq("a".attr.int, "b".attr.int, "c".attr.int), 1.to(10).map(_ => Row(1, 2, 3)) )
val query = GlobalLimit(100, testRelation)
```
We can remove this `GlobalLimit`.

### Why are the changes needed?

Further optimize the query.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Unit test.

### CODE REVIEW

**Comment by wangyum** [comment]
cc @cloud-fan This is a part of https://github.com/apache/spark/pull/31691. To make that pr more clear.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala [hunk 1]
```diff
@@ -1559,7 +1559,8 @@ object PushPredicateThroughJoin extends Rule[LogicalPlan] with PredicateHelper {
 /**
  * This rule optimizes Limit operators by:
  * 1. Eliminate [[Limit]] operators if it's child max row <= limit.
- * 2. Combines two adjacent [[Limit]] operators into one, merging the
+ * 2. Eliminate [[GlobalLimit]] operators if it's child max row <= limit.
```
Comment (maropu) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala:1562]: nit: how about merging `1.` and `2.` like `Eliminate [[Limit]]/[[GlobalLimit]] ...`?

**Review by maropu — APPROVED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala (hunk 1 above)
Comment (wangyum) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala]: fixed.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @wangyum and @maropu .
Merged to master.

**Comment by cloud-fan** [comment]
late LGTM

**Pull request closed without merging**